File: scratch/kap_test.py
import urllib.request
import json

headers = {
    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36",
    "Accept": "application/json, text/plain, */*",
    "Referer": "https://www.kap.org.tr/tr/bulten/gunluk",
}

# 1. Fetch recent disclosures from KAP
url = "https://www.kap.org.tr/tr/api/disclosures"
req = urllib.request.Request(url, headers=headers)

try:
    with urllib.request.urlopen(req) as resp:
        data = json.loads(resp.read().decode("utf-8"))
        print(f"Total disclosures fetched: {len(data)}")
        
        # Filter for Portföy Dağılım Raporu (PDR)
        pdr_list = [d for d in data if "Portföy Dağılım" in (d.get("title") or "") or "Portföy Dağılım" in (d.get("summary") or "")]
        print(f"PDR count in recent disclosures: {len(pdr_list)}")
        
        if pdr_list:
            sample = pdr_list[0]
            print("Sample PDR:", json.dumps(sample, indent=2, ensure_ascii=False))
            idx = sample.get("disclosureIndex")
            print(f"Fetching detail for index: {idx}")
            
            detail_url = f"https://www.kap.org.tr/tr/api/disclosure/detail/{idx}"
            req2 = urllib.request.Request(detail_url, headers=headers)
            with urllib.request.urlopen(req2) as resp2:
                detail_data = json.loads(resp2.read().decode("utf-8"))
                print("Detail keys:", list(detail_data.keys()))
                # Save to file
                with open("scratch/kap_sample.json", "w", encoding="utf-8") as f:
                    json.dump(detail_data, f, indent=2, ensure_ascii=False)
                print("Saved scratch/kap_sample.json")
        else:
            # Let's inspect some disclosure titles
            print("First 5 disclosure titles:")
            for d in data[:5]:
                print("-", d.get("basic", {}).get("companyName"), "|", d.get("basic", {}).get("title"), "|", d.get("basic", {}).get("disclosureType"))
except Exception as e:
    print("Error:", e)

File: scratch/kap_test.mjs
import fs from "fs";

const headers = {
  "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36",
  "Accept": "application/json, text/plain, */*",
  "Referer": "https://www.kap.org.tr/tr/bulten/gunluk",
};

async function main() {
  console.log("Fetching KAP disclosures...");
  try {
    const res = await fetch("https://www.kap.org.tr/tr/api/disclosures", { headers });
    if (!res.ok) {
      console.error("HTTP error:", res.status);
      return;
    }
    const data = await res.json();
    console.log(`Fetched ${data.length} disclosures from KAP.`);
    
    // Look for Portföy Dağılım Raporu
    const pdrList = data.filter((d) => {
      const title = d.basic?.title || d.title || "";
      const summary = d.basic?.summary || d.summary || "";
      const type = d.basic?.disclosureType || d.disclosureType || "";
      return title.includes("Portföy") || summary.includes("Portföy") || type.includes("PDR");
    });
    
    console.log(`Found ${pdrList.length} PDR notifications.`);
    if (pdrList.length > 0) {
      console.log("Sample PDR:", JSON.stringify(pdrList[0], null, 2));
      const idx = pdrList[0].disclosureIndex || pdrList[0].basic?.disclosureIndex;
      if (idx) {
        console.log(`Fetching detail for disclosureIndex: ${idx}`);
        const detRes = await fetch(`https://www.kap.org.tr/tr/api/disclosure/detail/${idx}`, { headers });
        const detJson = await detRes.json();
        fs.writeFileSync("scratch/kap_detail_sample.json", JSON.stringify(detJson, null, 2));
        console.log("Saved scratch/kap_detail_sample.json");
      }
    } else {
      console.log("Sample 3 disclosures:", JSON.stringify(data.slice(0, 3), null, 2));
    }
  } catch (err) {
    console.error("Fetch error:", err);
  }
}

main();
